solidity/dashboard/src/sagas/staking.js

```diff
@@ -166,15 +182,92 @@ function* fetchTopUps() {
     }
 
     yield put({ type: "staking/fetch_top_ups_start" })
-    const { delegations, undelegations } = yield select(
+    const { delegations, undelegations, initializationPeriod } = yield select(
       (state) => state.staking
     )
     const operators = [...undelegations, ...delegations].map(
       ({ operatorAddress }) => operatorAddress
     )
     const topUps = yield call(fetchAvailableTopUps, operators)
-    yield put({ type: "staking/fetch_top_ups_success", payload: topUps })
+    yield put({
+      type: "staking/fetch_top_ups_success",
+      payload: topUps.map((_) => ({
+        ..._,
+        readyToBeCommitted: isTopUpReadyToBeCommitted(_, initializationPeriod),
+      })),
+    })
   } catch (error) {
     yield* logError("staking/fetch_top_ups_failure", error)
   }
 }
+
+export function* watchTopUpReadyToBeCommitted() {
+  // Waiting for top-ups data.
+  yield take("staking/fetch_top_ups_success")
+  yield call(notifyTopUpReadyToBeCommitted)
+
+  while (true) {
+    // Every 5 minutes.
+    yield delay(moment.duration(5, "minutes").asMilliseconds())
+    yield call(notifyTopUpReadyToBeCommitted)
+  }
+}
+
+function* notifyTopUpReadyToBeCommitted() {
+  const topUps = yield select((state) => state.staking.topUps)
+  const initializationPeriod = yield select(
+    (state) => state.staking.initializationPeriod
+  )
+
+  const topUpsReadyToCommit = topUps.filter((topUp) =>
+    isTopUpReadyToBeCommitted(topUp, initializationPeriod)
+  )
+
+  if (isEmptyArray(topUpsReadyToCommit)) {
+    return
+  }
+
+  yield put({
+    type: "staking/top_ups_ready_to_be_committed",
+    payload: topUpsReadyToCommit,
+  })
+
+  const { delegations } = yield select((state) => state.staking)
+
+  let isFromLiquidTokens = false
+  for (const { operatorAddress } of topUpsReadyToCommit) {
+    if (!isFromLiquidTokens) {
+      isFromLiquidTokens = delegations.some(
+        (_) =>
+          isSameEthAddress(_.operatorAddress, operatorAddress) && !_.isFromGrant
+      )
+    }
+    const stake = delegations.find(
+      (_) =>
+        isSameEthAddress(_.operatorAddress, operatorAddress) && _.isFromGrant
+    )
+
+    if (stake) {
+      yield put(
+        showMessage({
+          messageType: messageType.TOP_UP_READY_TO_BE_COMMITTED,
+          messageProps: {
+            sticky: true,
+            grantId: stake.grantId,
+          },
+        })
+      )
+    }
+  }
```

Comment:
There is an issue with top up notifications - there are too many of them after a while, if user is not closing them 😛 

So i kept the site working in the background for a while, with my wallet connected, and this is what happened:
![image](https://user-images.githubusercontent.com/40306834/109619692-61420100-7b39-11eb-8dc3-4040d6fe8888.png)


We had [similar problem with liquidity rewards notification](https://github.com/keep-network/keep-core/pull/2346).

I think we only need one message for the top-ups, and we don't need to display it if it's already displayed. So i propose similar solution that i applied:

```
const displayedMessages = yield select((state) => state.messages)
const topUpReadeToBeCommittedNotificationAlreadyDisplayed = displayedMessages.some(
  (message) => {
    return message.messageType === messageType. TOP_UP_READY_TO_BE_COMMITTED
  }
)
```

and then display the message `if (topUpReadeToBeCommittedNotificationAlreadyDisplayed === true)`